modules/tensor_mechanics/src/actions/InclinedNoDisplacementBCAction.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InclinedNoDisplacementBCAction.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+
+registerMooseAction("TensorMechanicsApp", InclinedNoDisplacementBCAction, "add_bc");
+
+template <>
+InputParameters
+validParams<InclinedNoDisplacementBCAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up inclined no displacement boundary conditions");
+
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs from the mesh where the pressure will be applied");
+
+  params.addParam<VariableName>("disp_x", "The x displacement");
+  params.addParam<VariableName>("disp_y", "The y displacement");
+  params.addParam<VariableName>("disp_z", "The z displacement");
+
+  params.addParam<std::vector<VariableName>>(
+      "displacements",
+      "The displacements appropriate for the simulation geometry and coordinate system");
+
+  params.addParam<std::vector<AuxVariableName>>("save_in_disp_x",
+                                                "The save_in variables for x displacement");
+  params.addParam<std::vector<AuxVariableName>>("save_in_disp_y",
+                                                "The save_in variables for y displacement");
+  params.addParam<std::vector<AuxVariableName>>("save_in_disp_z",
+                                                "The save_in variables for z displacement");
+  params.addRequiredParam<Real>("penalty", "Penalty parameter");
+  return params;
+}
+
+InclinedNoDisplacementBCAction::InclinedNoDisplacementBCAction(const InputParameters & params)
+  : Action(params)
+{
+  _save_in_vars.push_back(getParam<std::vector<AuxVariableName>>("save_in_disp_x"));
+  _save_in_vars.push_back(getParam<std::vector<AuxVariableName>>("save_in_disp_y"));
+  _save_in_vars.push_back(getParam<std::vector<AuxVariableName>>("save_in_disp_z"));
```

Comment:
Without a default, doesn't this error when it's not provided?

Reply:
I tested it and it does not error :-)

_save_in_vars[i] will be empty if it is not provided.